packages/riipen-ui/src/components/ButtonIcon.test.jsx

```diff
@@ -0,0 +1,237 @@
+    const wrapper = mount(<ButtonIcon>children</ButtonIcon>);
+
+    const component = wrapper.find("ButtonIcon");
+    expect(component.props().classes).toEqual([]);
+    expect(component.props().color).toEqual("default");
+    expect(component.props().disabled).toEqual(false);
+    expect(component.props().size).toEqual("medium");
+  });
+
+  describe("children prop", () => {
+    it("displays given children", () => {
+      const child = "children";
+
+      const wrapper = mount(<ButtonIcon>{child}</ButtonIcon>);
+
+      expect(wrapper.find("ButtonIcon").contains(child)).toEqual(true);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("applies custom classes", () => {
+      const classes = ["classOne"];
+
+      const wrapper = mount(
+        <ButtonIcon classes={classes}>children</ButtonIcon>
+      );
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("color prop", () => {
+    it("sets color class name with a valid color", () => {
+      const color = "muted";
+
+      const wrapper = mount(<ButtonIcon color={color}>children</ButtonIcon>);
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .hasClass(color)
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<ButtonIcon color="pink">children</ButtonIcon>);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("component prop", () => {
+    it("sets component with valid custom component", () => {
+      const component = "span";
+
+      const wrapper = mount(
+        <ButtonIcon component={component}>children</ButtonIcon>
+      );
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("sets component with 'button' when component prop is not provided", () => {
+      const wrapper = mount(<ButtonIcon>children</ButtonIcon>);
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .name()
+      ).toEqual("button");
+    });
+
+    it("sets component with 'a' when href is defined", () => {
+      const href = "www.google.com";
+
+      const wrapper = mount(<ButtonIcon href={href}>children</ButtonIcon>);
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .name()
+      ).toEqual("a");
+    });
+
+    it("gives an error when given an invalid component", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcd";
+
+      mount(<ButtonIcon component={component}>children</ButtonIcon>);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("disabled prop", () => {
```

Comment:
I think we are also missing the test "sets disabled on button when disabled prop is true".

Reply:
Is that the same thing as "applies disabled class when disabled prop is true"? It supposed to test line 52 in ButtonIcon.

Reply:
Sorry, I don't think I was clear.

    The test you have in this file on line 137 tests line 35 in the source file
    We should add a test to test line 53 in the source file.